Unknown diagram extension causes exception

## Example

```markdown
# Strings

There is an Element.
![element](foo.gif)

```

## Example

```
1 error generated.
failed to execute phase 'CodeGenerator' due to exception:
java.lang.IllegalStateException: java.lang.IllegalStateException: invalid file name 'foo.gif' - unsupported extension
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at java.util.concurrent.ForkJoinTask.getThrowableException(ForkJoinTask.java:593)
	at java.util.concurrent.ForkJoinTask.reportException(ForkJoinTask.java:677)
	at java.util.concurrent.ForkJoinTask.invoke(ForkJoinTask.java:735)
	at java.util.stream.ForEachOps$ForEachOp.evaluateParallel(ForEachOps.java:159)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateParallel(ForEachOps.java:173)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:233)
	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:485)
	at java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:650)
	at org.fulib.scenarios.visitor.codegen.CodeGenerator.visit(CodeGenerator.java:52)
	at org.fulib.scenarios.ast.CompilationContext.accept(CompilationContext.java:24)
	at org.fulib.scenarios.tool.ScenarioCompiler.run(ScenarioCompiler.java:145)
	at org.fulib.scenarios.tool.ScenarioCompiler.run(ScenarioCompiler.java:129)
	at org.fulib.scenarios.tool.ScenarioCompiler.run(ScenarioCompiler.java:97)
	at org.fulib.webapp.tool.Tools.scenarioc(Tools.java:83)
	at org.fulib.webapp.tool.Tools.genCompileRun(Tools.java:161)
	at org.fulib.webapp.tool.RunCodeGen.run(RunCodeGen.java:152)
	at org.fulib.webapp.tool.RunCodeGen.handle(RunCodeGen.java:54)
	at spark.RouteImpl$1.handle(RouteImpl.java:72)
	at spark.http.matching.Routes.execute(Routes.java:61)
	at spark.http.matching.MatcherFilter.doFilter(MatcherFilter.java:134)
	at spark.embeddedserver.jetty.JettyHandler.doHandle(JettyHandler.java:50)
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1584)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
	at org.eclipse.jetty.server.Server.handle(Server.java:501)
	at org.eclipse.jetty.server.HttpChannel.lambda$handle$1(HttpChannel.java:383)
	at org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:556)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:375)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:272)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103)
	at org.eclipse.jetty.io.ChannelEndPoint$1.run(ChannelEndPoint.java:104)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:336)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:313)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:171)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:129)
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:375)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:806)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:938)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.IllegalStateException: invalid file name 'foo.gif' - unsupported extension
	at org.fulib.scenarios.visitor.codegen.SentenceGenerator.visit(SentenceGenerator.java:224)
	at org.fulib.scenarios.visitor.codegen.SentenceGenerator.visit(SentenceGenerator.java:20)
	at org.fulib.scenarios.ast.sentence.DiagramSentence.accept(DiagramSentence.java:22)
	at org.fulib.scenarios.visitor.codegen.SentenceGenerator.visit(SentenceGenerator.java:35)
	at org.fulib.scenarios.visitor.codegen.SentenceGenerator.visit(SentenceGenerator.java:20)
	at org.fulib.scenarios.ast.sentence.SentenceList.accept(SentenceList.java:20)
	at org.fulib.scenarios.visitor.codegen.DeclGenerator.visit(DeclGenerator.java:115)
	at org.fulib.scenarios.visitor.codegen.DeclGenerator.visit(DeclGenerator.java:12)
	at org.fulib.scenarios.ast.decl.MethodDecl.accept(MethodDecl.java:30)
	at org.fulib.scenarios.visitor.codegen.DeclGenerator.visit(DeclGenerator.java:38)
	at org.fulib.scenarios.visitor.codegen.DeclGenerator.visit(DeclGenerator.java:12)
	at org.fulib.scenarios.ast.decl.ClassDecl.accept(ClassDecl.java:37)
	at org.fulib.scenarios.visitor.codegen.CodeGenerator.visit(CodeGenerator.java:318)
	at org.fulib.scenarios.visitor.codegen.CodeGenerator.visit(CodeGenerator.java:32)
	at org.fulib.scenarios.ast.ScenarioFile.accept(ScenarioFile.java:29)
	at org.fulib.scenarios.visitor.codegen.CodeGenerator.populateTests(CodeGenerator.java:164)
	at org.fulib.scenarios.visitor.codegen.CodeGenerator.visit(CodeGenerator.java:149)
	at org.fulib.scenarios.visitor.codegen.CodeGenerator.visit(CodeGenerator.java:32)
	at org.fulib.scenarios.ast.ScenarioGroup.accept(ScenarioGroup.java:26)
	at org.fulib.scenarios.visitor.codegen.CodeGenerator.lambda$visit$0(CodeGenerator.java:56)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.util.concurrent.ConcurrentHashMap$ValueSpliterator.forEachRemaining(ConcurrentHashMap.java:3566)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:482)
	at java.util.stream.ForEachOps$ForEachTask.compute(ForEachOps.java:290)
	at java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:731)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:175)

```

--- a/src/main/java/org/fulib/scenarios/tool/Config.java
+++ b/src/main/java/org/fulib/scenarios/tool/Config.java
@@ -17,10 +17,19 @@ public class Config
    private List<String> classpath = new ArrayList<>();
    private Set<String>  imports   = new HashSet<>();
 
+   private final Map<String, String> diagramHandlers = new HashMap<>();
+
    {
       // default imports
       this.imports.add("java.lang");
       this.imports.add("org.fulib.mockups");
+
+      this.diagramHandlers.put(".svg", "import(org.fulib.tools.FulibTools).objectDiagrams().dumpSVG(%s, %s)");
+      this.diagramHandlers.put(".html.png", "import(org.fulib.mockups.FulibMockups).mockupTool().dump(%s, %s)");
+      this.diagramHandlers.put(".png", "import(org.fulib.tools.FulibTools).objectDiagrams().dumpPng(%s, %s)");
+      this.diagramHandlers.put(".yaml", "import(org.fulib.tools.FulibTools).objectDiagrams().dumpYaml(%s, %s)");
+      this.diagramHandlers.put(".html", "import(org.fulib.scenarios.MockupTools).htmlTool().dump(%s, %s)");
+      this.diagramHandlers.put(".txt", "import(org.fulib.scenarios.MockupTools).htmlTool().dumpToString(%s, %s)");
    }
 
    private Set<String> decoratorClasses = new TreeSet<>();
@@ -77,6 +86,11 @@ public class Config
    public Set<String> getDecoratorClasses()
    {
       return this.decoratorClasses;
+   }
+
+   public String getDiagramHandler(String extension)
+   {
+      return this.diagramHandlers.get(extension);
    }
 
    public boolean isGenerateTables()
@@ -194,6 +208,15 @@ public class Config
       options.addOption(
          new Option(null, "marker-end-columns", false, "include the column number where a marker ends in the output"));
 
+      options.addOption(Option
+                           .builder()
+                           .longOpt("diagram-handlers")
+                           .argName("<extension>=<method>")
+                           .numberOfArgs(2)
+                           .valueSeparator()
+                           .desc("generate diagrams with the extension using the handler method")
+                           .build());
+
       return options;
    }
 
@@ -226,6 +249,11 @@ public class Config
       if (decoratorClasses != null)
       {
          Collections.addAll(this.decoratorClasses, decoratorClasses);
+      }
+
+      for (final Map.Entry<Object, Object> entry : cmd.getOptionProperties("diagram-handlers").entrySet())
+      {
+         this.diagramHandlers.put(entry.getKey().toString(), entry.getValue().toString());
       }
    }
 }

--- a/src/main/java/org/fulib/scenarios/tool/Config.java
+++ b/src/main/java/org/fulib/scenarios/tool/Config.java
@@ -88,6 +88,11 @@ public class Config
       return this.decoratorClasses;
    }
 
+   public Set<String> getDiagramHandlerExtensions()
+   {
+      return this.diagramHandlers.keySet();
+   }
+
    public String getDiagramHandler(String extension)
    {
       return this.diagramHandlers.get(extension);

--- a/src/main/java/org/fulib/scenarios/tool/Config.java
+++ b/src/main/java/org/fulib/scenarios/tool/Config.java
@@ -11,11 +11,13 @@ public class Config
 {
    // =============== Fields ===============
 
-   private String       modelDir;
-   private String       testDir;
-   private List<String> inputDirs = new ArrayList<>();
-   private List<String> classpath = new ArrayList<>();
-   private Set<String>  imports   = new HashSet<>();
+   private String modelDir;
+   private String testDir;
+   private final List<String> inputDirs = new ArrayList<>();
+   private final List<String> classpath = new ArrayList<>();
+   private final Set<String> imports = new HashSet<>();
+
+   private final Set<String> decoratorClasses = new TreeSet<>();
 
    private final Map<String, String> diagramHandlers = new HashMap<>();
 
@@ -33,8 +35,6 @@ public class Config
       this.diagramHandlers.put(".mockup.html", "import(org.fulib.scenarios.MockupTools).htmlTool().dumpMockup(%s)");
       this.diagramHandlers.put(".txt", "import(org.fulib.scenarios.MockupTools).htmlTool().dumpToString(%s, %s)");
    }
-
-   private Set<String> decoratorClasses = new TreeSet<>();
 
    private boolean generateTables;
 

--- a/src/main/java/org/fulib/scenarios/tool/Config.java
+++ b/src/main/java/org/fulib/scenarios/tool/Config.java
@@ -24,10 +24,10 @@ public class Config
       this.imports.add("java.lang");
       this.imports.add("org.fulib.mockups");
 
-      this.diagramHandlers.put(".svg", "import(org.fulib.tools.FulibTools).objectDiagrams().dumpSVG(%s, %s)");
+      this.diagramHandlers.put(".svg", "import(org.fulib.FulibTools).objectDiagrams().dumpSVG(%s, %s)");
       this.diagramHandlers.put(".html.png", "import(org.fulib.mockups.FulibMockups).mockupTool().dump(%s, %s)");
-      this.diagramHandlers.put(".png", "import(org.fulib.tools.FulibTools).objectDiagrams().dumpPng(%s, %s)");
-      this.diagramHandlers.put(".yaml", "import(org.fulib.tools.FulibTools).objectDiagrams().dumpYaml(%s, %s)");
+      this.diagramHandlers.put(".png", "import(org.fulib.FulibTools).objectDiagrams().dumpPng(%s, %s)");
+      this.diagramHandlers.put(".yaml", "import(org.fulib.FulibTools).objectDiagrams().dumpYaml(%s, %s)");
       this.diagramHandlers.put(".html", "import(org.fulib.scenarios.MockupTools).htmlTool().dump(%s, %s)");
       this.diagramHandlers.put(".txt", "import(org.fulib.scenarios.MockupTools).htmlTool().dumpToString(%s, %s)");
    }

--- a/src/main/java/org/fulib/scenarios/tool/Config.java
+++ b/src/main/java/org/fulib/scenarios/tool/Config.java
@@ -93,6 +93,23 @@ public class Config
       return this.diagramHandlers.get(extension);
    }
 
+   public String getDiagramHandlerFromFile(String fileName)
+   {
+      int dotIndex = -1;
+
+      while ((dotIndex = fileName.indexOf('.', dotIndex + 1)) >= 0)
+      {
+         final String extension = fileName.substring(dotIndex);
+         final String handler = this.getDiagramHandler(extension);
+         if (handler != null)
+         {
+            return handler;
+         }
+      }
+
+      return null;
+   }
+
    public boolean isGenerateTables()
    {
       return this.generateTables;

--- a/src/main/java/org/fulib/scenarios/tool/Config.java
+++ b/src/main/java/org/fulib/scenarios/tool/Config.java
@@ -28,7 +28,9 @@ public class Config
       this.diagramHandlers.put(".html.png", "import(org.fulib.mockups.FulibMockups).mockupTool().dump(%s, %s)");
       this.diagramHandlers.put(".png", "import(org.fulib.FulibTools).objectDiagrams().dumpPng(%s, %s)");
       this.diagramHandlers.put(".yaml", "import(org.fulib.FulibTools).objectDiagrams().dumpYaml(%s, %s)");
-      this.diagramHandlers.put(".html", "import(org.fulib.scenarios.MockupTools).htmlTool().dump(%s, %s)");
+      this.diagramHandlers.put(".html", "import(org.fulib.scenarios.MockupTools).htmlTool().dumpScreen(%s, %s)");
+      this.diagramHandlers.put(".tables.html", "import(org.fulib.scenarios.MockupTools).htmlTool().dumpTables(%s, %s)");
+      this.diagramHandlers.put(".mockup.html", "import(org.fulib.scenarios.MockupTools).htmlTool().dumpMockup(%s)");
       this.diagramHandlers.put(".txt", "import(org.fulib.scenarios.MockupTools).htmlTool().dumpToString(%s, %s)");
    }
 

--- a/src/main/java/org/fulib/scenarios/visitor/codegen/SentenceGenerator.java
+++ b/src/main/java/org/fulib/scenarios/visitor/codegen/SentenceGenerator.java
@@ -1,5 +1,6 @@
 package org.fulib.scenarios.visitor.codegen;
 
+import org.apache.commons.text.StringEscapeUtils;
 import org.fulib.StrUtil;
 import org.fulib.scenarios.ast.NamedExpr;
 import org.fulib.scenarios.ast.decl.Decl;
@@ -186,62 +187,26 @@ public enum SentenceGenerator implements Sentence.Visitor<CodeGenDTO, Object>
       final String sourceDir = par.group.getSourceDir();
       final String packageDir = par.group.getPackageDir();
       final String fileName = diagramSentence.getFileName();
-      String target = (sourceDir + "/" + packageDir + "/" + fileName).replace('\\', '/');
+      final String target = (sourceDir + "/" + packageDir + "/" + fileName).replace('\\', '/');
 
-      final int dotIndex = fileName.lastIndexOf('.');
-      if (dotIndex < 0)
+      final String diagramHandler = par.config.getDiagramHandlerFromFile(fileName);
+      final String targetLiteral = '"' + StringEscapeUtils.escapeJava(target) + '"';
+
+      final StringBuilder oldBuilder = par.bodyBuilder;
+      final String objectExpr;
+      par.bodyBuilder = new StringBuilder();
+      try
       {
-         throw new IllegalStateException("invalid file name '" + fileName + "' - missing extension");
+         diagramSentence.getObject().accept(ExprGenerator.FLAT, par);
+         objectExpr = par.bodyBuilder.toString();
       }
-
-      final String extension = fileName.substring(dotIndex).toLowerCase();
-      final String toolClass;
-      final String toolMethod;
-      switch (extension)
+      finally
       {
-      case ".svg":
-         toolClass = "org.fulib.FulibTools";
-         toolMethod = "FulibTools.objectDiagrams().dumpSVG";
-         break;
-      case ".png":
-         if (fileName.endsWith(".html.png"))
-         {
-            toolClass = "org.fulib.mockups.FulibMockups";
-            toolMethod = "FulibMockups.mockupTool().dump";
-         }
-         else
-         {
-            toolClass = "org.fulib.FulibTools";
-            toolMethod = "FulibTools.objectDiagrams().dumpPng";
-         }
-         break;
-      case ".yaml":
-         toolClass = "org.fulib.FulibTools";
-         toolMethod = "FulibTools.objectDiagrams().dumpYaml";
-         break;
-      case ".html":
-         toolClass = "org.fulib.scenarios.MockupTools";
-         toolMethod = "MockupTools.htmlTool().dump";
-         break;
-      case ".txt":
-         toolClass = "org.fulib.scenarios.MockupTools";
-         toolMethod = "MockupTools.htmlTool().dumpToString";
-
-         break;
-      default:
-         throw new IllegalStateException("invalid file name '" + fileName + "' - unsupported extension");
+         par.bodyBuilder = oldBuilder;
       }
-
-      par.addImport(toolClass);
 
       par.emitIndent();
-
-      // method call
-      par.bodyBuilder.append(toolMethod).append('(');
-      par.emitStringLiteral(target);
-      par.bodyBuilder.append(", ");
-      diagramSentence.getObject().accept(ExprGenerator.FLAT, par);
-      par.bodyBuilder.append(");\n");
+      par.bodyBuilder.append(String.format(diagramHandler, targetLiteral, objectExpr)).append(";\n");
 
       return null;
    }

--- a/src/main/java/org/fulib/scenarios/visitor/codegen/SentenceGenerator.java
+++ b/src/main/java/org/fulib/scenarios/visitor/codegen/SentenceGenerator.java
@@ -205,8 +205,11 @@ public enum SentenceGenerator implements Sentence.Visitor<CodeGenDTO, Object>
          par.bodyBuilder = oldBuilder;
       }
 
-      par.emitIndent();
-      par.bodyBuilder.append(String.format(diagramHandler, targetLiteral, objectExpr)).append(";\n");
+      if (diagramHandler != null)
+      {
+         par.emitIndent();
+         par.bodyBuilder.append(String.format(diagramHandler, targetLiteral, objectExpr)).append(";\n");
+      }
 
       return null;
    }

--- a/src/main/java/org/fulib/scenarios/visitor/resolve/SentenceResolver.java
+++ b/src/main/java/org/fulib/scenarios/visitor/resolve/SentenceResolver.java
@@ -24,6 +24,7 @@ import org.fulib.scenarios.ast.type.PrimitiveType;
 import org.fulib.scenarios.ast.type.Type;
 import org.fulib.scenarios.diagnostic.Marker;
 import org.fulib.scenarios.diagnostic.Position;
+import org.fulib.scenarios.tool.Config;
 import org.fulib.scenarios.visitor.ExtractClassDecl;
 import org.fulib.scenarios.visitor.Namer;
 import org.fulib.scenarios.visitor.TypeConversion;
@@ -299,6 +300,20 @@ public enum SentenceResolver implements Sentence.Visitor<Scope, Sentence>
    public Sentence visit(DiagramSentence diagramSentence, Scope par)
    {
       diagramSentence.setObject(diagramSentence.getObject().accept(ExprResolver.INSTANCE, par));
+
+      final Config config = DeclResolver.getEnclosingClass(par).getGroup().getContext().getConfig();
+
+      final String fileName = diagramSentence.getFileName();
+      final String diagramHandler = config.getDiagramHandlerFromFile(fileName);
+      if (diagramHandler == null)
+      {
+         final Position position = diagramSentence.getPosition();
+         final Marker error = error(position, "diagram.filename.extension.unsupported", fileName);
+         error.note(
+            note(position, "diagram.filename.extension.hint", new TreeSet<>(config.getDiagramHandlerExtensions())));
+         par.report(error);
+      }
+
       return diagramSentence;
    }
 

--- a/src/main/resources/org/fulib/scenarios/diagnostic/messages.properties
+++ b/src/main/resources/org/fulib/scenarios/diagnostic/messages.properties
@@ -91,6 +91,9 @@ inherit.supertype.external=cannot inherit from external class '%s'
 inherit.conflict=cannot change supertype of '%s' which already inherits from '%s'
 inherit.declaration.first='%s' was first declared to inherit from '%s' here
 
+diagram.filename.extension.unsupported=unsupported diagram file name extension '%s'
+diagram.filename.extension.hint=extension must be one of %s
+
 # --------------- Constraints ---------------
 
 match.no.roots=match has no root objects - no objects are in scope or declared with 'on ...'

--- a/src/test/invalid_scenarios/lang/Diagrams.md
+++ b/src/test/invalid_scenarios/lang/Diagrams.md
@@ -6,5 +6,5 @@ There is the Object foo.
 <!-- ^^^^^^^^^^^^^^^
 error: unsupported diagram file name extension 'foo.gif' [diagram.filename.extension.unsupported]
      ^^^^^^^^^^^^^^^
-note: extension must be one of [.html, .html.png, .png, .svg, .txt, .yaml] [diagram.filename.extension.hint]
+note: extension must be one of [.html, .html.png, .mockup.html, .png, .svg, .tables.html, .txt, .yaml] [diagram.filename.extension.hint]
 -->

--- a/src/test/invalid_scenarios/lang/Diagrams.md
+++ b/src/test/invalid_scenarios/lang/Diagrams.md
@@ -1,0 +1,10 @@
+# Invalid Diagram Extensions
+
+There is the Object foo.
+
+     ![foo](foo.gif)
+<!-- ^^^^^^^^^^^^^^^
+error: unsupported diagram file name extension 'foo.gif' [diagram.filename.extension.unsupported]
+     ^^^^^^^^^^^^^^^
+note: extension must be one of [.html, .html.png, .png, .svg, .txt, .yaml] [diagram.filename.extension.hint]
+-->

--- a/src/test/java/org/fulib/scenarios/tool/ConfigTest.java
+++ b/src/test/java/org/fulib/scenarios/tool/ConfigTest.java
@@ -23,8 +23,20 @@ public class ConfigTest
    @Test
    public void diagramHandlers() throws ParseException
    {
-      final Config config = parse("--diagram-handlers", ".gif=GifTool.dump", "--diagram-handlers", ".html=HtmlTool.dump");
+      final String[] args = {
+         "--diagram-handlers",
+         ".gif=GifTool.dump",
+         "--diagram-handlers",
+         ".html=HtmlTool.dump",
+         "--diagram-handlers",
+         ".html.gif=HtmlGifTool.dump",
+      };
+      final Config config = parse(args);
 
       MatcherAssert.assertThat(config.getDiagramHandler(".gif"), CoreMatchers.equalTo("GifTool.dump"));
+      MatcherAssert.assertThat(config.getDiagramHandlerFromFile("example.gif"), CoreMatchers.equalTo("GifTool.dump"));
+      MatcherAssert.assertThat(config.getDiagramHandlerFromFile("example.html"), CoreMatchers.equalTo("HtmlTool.dump"));
+      MatcherAssert.assertThat(config.getDiagramHandlerFromFile("example.html.gif"),
+                               CoreMatchers.equalTo("HtmlGifTool.dump"));
    }
 }

--- a/src/test/java/org/fulib/scenarios/tool/ConfigTest.java
+++ b/src/test/java/org/fulib/scenarios/tool/ConfigTest.java
@@ -1,0 +1,30 @@
+package org.fulib.scenarios.tool;
+
+import org.apache.commons.cli.*;
+import org.hamcrest.CoreMatchers;
+import org.hamcrest.MatcherAssert;
+import org.junit.Test;
+
+public class ConfigTest
+{
+   private static Config parse(String... args) throws ParseException
+   {
+      final Config config = new Config();
+      final Options options = config.createOptions();
+
+      final CommandLineParser parser = new DefaultParser();
+      final CommandLine commandLine = parser.parse(options, args);
+
+      config.readOptions(commandLine);
+
+      return config;
+   }
+
+   @Test
+   public void diagramHandlers() throws ParseException
+   {
+      final Config config = parse("--diagram-handlers", ".gif=GifTool.dump", "--diagram-handlers", ".html=HtmlTool.dump");
+
+      MatcherAssert.assertThat(config.getDiagramHandler(".gif"), CoreMatchers.equalTo("GifTool.dump"));
+   }
+}

